src/js/components/invoice/InvoiceListFilters.jsx

```diff
@@ -109,21 +110,29 @@ const InvoiceListFilters = ({
   fetchTypeCodes,
 }) => {
   const [defaultValues, setDefaultValues] = useState({});
+  const prevLocation = usePrevious(currentLocation);
+  const [initiallyFetched, setInitiallyFetched] = useState(false);
 
   useEffect(() => {
-    const initialEmptyValues = Object.keys(filterFields).reduce((acc, key) => {
-      if (!acc[key]) return { ...acc, [key]: '' };
-      return acc;
-    }, {});
-    setDefaultValues({
-      ...initialEmptyValues,
-      buyerOrganization: {
-        id: currentLocation?.organization?.id,
-        value: currentLocation?.organization?.id,
-        name: currentLocation?.organization?.name,
-        label: currentLocation?.organization?.name,
-      },
-    });
+    // Avoid unnecessary re-fetches if getAppContext triggers fetching session info
+    // but currentLocation doesn't change
+    // eslint-disable-next-line max-len
+    if ((!initiallyFetched && currentLocation?.id) || (prevLocation && prevLocation.id !== currentLocation?.id)) {
```

Comment:
it had to be done like this because: 
you might have a situation when you have currentLocation in store and go to another list page -
1.  if we left only 
```
if (currentLocation.id)
```
we would refetch twice - once having already stored location and second time when the getAppContext made a refetch.
2. if we left only
```
if (prevLocation.id !== currentLocation.id) 
```
we wouldn't make a fetch if we have currentLocation already stored, because the `prevLocation` would be the `currentLocation` (that was in store) and the `currentLocation` (being fetched from getAppContext) would be the same what `prevLocation` so we wouldn't make a single fetch.
That's why I added `initiallyFetched`, so if we have a situation where we have `currentLocation` in store, we make only **one** fetch, because we don't care about the `currentLocation` (the same) being fetched in the background from the `getAppContext`

Reply:
We want to remove the complexity if we can just get to a place where the redux hook is not invoked unnecessarily when the current location has not changed. Our suspicion is that the the app context (being a complex object) might be too complex for whatever is detecting whether the current location hsa changed. Or maybe I misread that. 

Either way, one solution is adding a new API endpoint to retrieve just the current location and user IDs so that we can provide a simpler object to go through the useEffect hook to determine if the location changed.

For the issue mentioned above re: refetching twice vs not fetching at all ... I think I understand, but I have no idea how to do deal with that.

@awalkowiak mentioned that he will try to download the branch locally to test it out and see if there's an easy solution